controllers/release/release_adapter.go

```diff
@@ -228,23 +229,41 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingIsCreated() (results.Operation
 // set it to a context from the workspace in the ReleasePlan if one is defined or to its own context if not.
 func (a *Adapter) EnsureTargetContextIsSet() (results.OperationResult, error) {
 	if a.targetContext == nil {
+		a.targetContext = a.context
+	}
+
+	// If the target namespace/workspace is not set at the release status we should get the target from the ReleasePlan
+	if a.release.Status.Target == (kcp.NamespaceReference{}) {
+		patch := client.MergeFrom(a.release.DeepCopy())
 		releasePlan, err := a.getReleasePlan()
 		if err != nil {
-			patch := client.MergeFrom(a.release.DeepCopy())
 			a.release.MarkInvalid(v1alpha1.ReleaseReasonReleasePlanValidationError, err.Error())
 			return results.RequeueOnErrorOrStop(a.client.Status().Patch(a.context, a.release, patch))
 		}
 
-		if releasePlan.Spec.Target.Workspace == "" {
-			a.logger.Info("Running in a traditional kubernetes environment")
+		a.release.Status.Target = releasePlan.Spec.Target
+		err = a.client.Status().Patch(a.context, a.release, patch)
+		if err != nil {
+			return results.RequeueWithError(err)
+		}
+
+		if a.release.Status.Target.Workspace == "" {
 			a.targetContext = a.context
 		} else {
+			a.targetContext = logicalcluster.WithCluster(a.context, logicalcluster.New(a.release.Status.Target.Workspace))
+			// If this is a non-KCP cluster and the target workspace is set we should stop the release
+			// otherwise it fails at the release_controller because it can not switch to a non existent
+			// context
+			if _, ok := logicalcluster.ClusterFromContext(a.context); !ok {
```

Comment:
hmmm. I think we need to separate this out. If this check fails on `a.context`, set in a non-KCP environment is true.

However, `ReleasePlan Workspace does not exist` is only true if this check fails for `a.targetContext`, not `a.context`. Maybe we have two separate checks? @davidmogar wdyt